common/src/main/java/io/netty/util/concurrent/AbstractScheduledEventExecutor.java

```diff
@@ -243,12 +248,22 @@ protected void validateScheduled(long amount, TimeUnit unit) {
         if (inEventLoop()) {
             scheduledTaskQueue().add(task.setId(nextTaskId++));
         } else {
-            executeScheduledRunnable(new Runnable() {
+            final long deadlineNanos = task.deadlineNanos();
+            final Runnable addToQueue = new Runnable() {
                 @Override
                 public void run() {
                     scheduledTaskQueue().add(task.setId(nextTaskId++));
                 }
-            }, true, task.deadlineNanos());
+            };
+            if (beforeScheduledTaskSubmitted(deadlineNanos)) {
+                execute(addToQueue);
+            } else {
+                lazyExecute(addToQueue);
```

Comment:
in theory, if I constantly write from outside the event loop fast enough the mpsc queue will blow up, right?

Reply:
The usage here is equivalent to the current behaviour, just avoids the extra wrapper Runnable. 

I gave my thoughts w.r.t general use of `lazyExecute` in response to your earlier question. Of course the method could be used to break things but there are lots of ways to break things. For example if you schedule a gazillion future tasks it will blow up the pending task queue (ignoring delayed execution). Or if you schedule tasks onto the event loop from multiple other threads faster than it can keep up with them, etc.

Reply:
@johnou is this resolved ?

Reply:
Well it's just a theory and I haven't tested it, also invoking size() on the mpsc to track pending tasks is probably expensive. I am happy to test it out with our application to see how it behaves.

Reply:
> invoking size() on the mpsc to track pending tasks is probably expensive

@johnou could you elaborate on this? I'm not sure how it applies to the changes in this PR. Also I don't think I would call it expensive if you are thinking of the JCTools impl that we use, especially when called from the consumer thread.

Reply:
I must be mixing it up with https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentLinkedDeque.html#size() where it's not a constant time operation.

Reply:
@johnou so can this be resolved ?

Reply:
Yes I'm happy with the explanation and fact it is marked unstableapi.